build-logic/documentation/src/main/groovy/gradlebuild/docs/FindBadMultiLangSnippets.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.docs;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Checks adoc files for reversed multi-language snippet order.
+ * Only rule enforced:
+ *   In a ==== ... ==== block, if there are at least two [.multi-language-sample] snippets,
+ *   flag when the first is GROOVY and the second is KOTLIN.
+ */
+@CacheableTask
+public abstract class FindBadMultiLangSnippets extends DefaultTask {
+
+    @InputDirectory
+    @PathSensitive(PathSensitivity.RELATIVE)
+    public abstract DirectoryProperty getDocumentationRoot();
+
+    @TaskAction
+    public void checkMultiLanguageSnippets() {
+        Map<File, List<Error>> errors = new TreeMap<>();
+
+        getDocumentationRoot().getAsFileTree().matching(spec -> spec.include("**/*.adoc")).forEach(file -> {
+            gatherBadSnippetsInFile(file, errors);
+        });
+
+        if (!errors.isEmpty()) {
```

Comment:
💅 Wondering what the DX were if this got implemented using the Problems API? https://docs.gradle.org/current/userguide/reporting_problems.html